plugin/adsk/scripts/AETemplateHelpers.py

```diff
@@ -35,4 +45,162 @@ def ProxyShapePayloads(proxyShape, loadAll):
+            okCaption = mel.eval('getMayaUsdString("kLoad")')
+            fileFilter = mel.eval('getMayaUsdString("kAllUsdFiles")') + ' (*.usd *.usda *.usdc *.usdz );;*.usd;;*.usda;;*.usdc;;*.usdz';
+            res = cmds.fileDialog2(caption=title, fileMode=1, ff=fileFilter, okc=okCaption)
+            if res and len(res) == 1:
+                debugMessage('    User picked USD file, setting file path attribute')
+                # Simply set the file path attribute. The proxy shape will load the file.
+                usdFileToLoad = res[0]
+                cmds.setAttr(filePathAttr, usdFileToLoad, type='string')
+                return True
+        elif newFilePath is not None:
+            # Instead of opening a file dialog to get the USD file, simply
+            # use the one provided as input.
+            # Note: it is important to check "not None" since the new file path can be
+            #       an empty string.
+            debugMessage('  New file to load provided, setting filepath="%s"' % newFilePath)
+            cmds.setAttr(filePathAttr, newFilePath, type='string')
+            return True
+    except Exception as e:
+        debugMessage('ProxyShapeFilePathChanged() - Error: %s' % str(e))
+        pass
+    return False
+
+def ProxyShapeFilePathRefresh(filePathAttr):
+    # Function called from the MayaUsd Proxy Shape template when the refresh
+    # button of the file path attribute custom control is clicked.
+    #
+    # Arguments:
+    # - filePathAttr: string representing the file path attribute.
+    #
+    # Return Value:
+    # - None
+    #
+    debugMessage('ProxyShapeFilePathRefresh(%s)' % filePathAttr)
+    try:
+        stageName, proxyStage = GetStageFromProxyShapeAttr(filePathAttr)
+
+        # Is any layer in the layerstack dirty?
+        # - If nothing is dirty, we simply reload.
+        # - If any layer is dirty pop a confirmation dialog to reload.
+        # Note: since we are file-backed, this will reload based on the file
+        #       which could have changed on disk.
+        if not IsProxyShapeLayerStackDirty(proxyStage):
+            debugMessage('  No dirty layers, calling UsdStage.Reload()')
+            proxyStage.Reload()
+        else:
+            kTitleFormat = mel.eval('getMayaUsdString("kDiscardStageEditsTitle")')
+            kMsgFormat = mel.eval('getMayaUsdString("kDiscardStageEditsReloadMsg")')
+            kYes = mel.eval('getMayaUsdString("kButtonYes")')
+            kNo = mel.eval('getMayaUsdString("kButtonNo")')
+            kTitle = cmds.format(kTitleFormat, stringArg=stageName)
+            # Our title is a little long, and the confirmDialog command is not making the
+            # dialog wide enough to show it all. So by telling the message string to not
+            # word-wrap, the dialog is forced wider.
+            filepath = cmds.getAttr(filePathAttr)
+            if filepath:
+                filename = os.path.basename(filepath)
```

Comment:
Does our proxy shape handle the asset resolver and if so would I need a special case here when its not actually a file path?

Reply:
No idea.

Reply:
I don't think we do anything special with the actual string attribute, it gets passed to USD to do something with it and it's there that things get resolved.

Reply:
Great.